### PR TITLE
[EDU-1628] - Spaces Live Cursors example

### PR DESCRIPTION
**Pull request by GregHolmes**
This PR contains the two code samples for the new Examples section we are creating. It contains a Typescript and a React version for Live Cursors with Spaces. Which will be rendered in the code sandbox live examples in the future, along with others.

The README within each directory /examples/spaces-live-cursors/javascript and /examples/spaces-live-cursors/react contains information on how to get these running.

[Jira epic ticket](https://ably.atlassian.net/browse/EDU-1541)
[Jira ticket](https://ably.atlassian.net/browse/EDU-1628).

### CODE REVIEW

**Review by ttypic — COMMENTED** [review]
LGTM

File: examples/spaces-live-cursors/page.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Live cursors for applications
+
+Enable members to see other member's cursors live in realtime within your application
+
+Live Cursors with Ably Spaces enables realtime tracking and display of user cursor positions within your application. This feature significantly enhances the collaborative experience by providing visual cues of other users' activities and focus areas in shared workspaces.
```
Comment (m-hulbert) [inline examples/spaces-live-cursors/page.md:5]: Mixture of capitalisation for live cursors; it should be lower case throughout.

File: examples/spaces-live-cursors/page.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Live cursors for applications
```
Comment (m-hulbert) [inline examples/spaces-live-cursors/page.md:1]: The title doesn't follow the same format as the Chat ones. Maybe something like "Tracking users' cursors"?

File: examples/spaces-live-cursors/page.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+Enable members to see other member's cursors live in realtime within your application
```
Comment (m-hulbert) [inline examples/spaces-live-cursors/page.md:3]:
```suggestion
Label and track the cursors of users in an application.
```

File: examples/spaces-live-cursors/page.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+Live Cursors with Ably Spaces enables realtime tracking and display of user cursor positions within your application. This feature significantly enhances the collaborative experience by providing visual cues of other users' activities and focus areas in shared workspaces.
+
+Live Cursors can add value to various collaborative applications. In online whiteboarding tools, they allow team members to see where others are pointing or drawing. In collaborative coding environments, live cursors can show which parts of the code different developers are currently working on, facilitating better coordination and reducing conflicts.
```
Comment (m-hulbert) [inline examples/spaces-live-cursors/page.md:7]:
```suggestion
Track and display the position of users' cursors within an application to provide visual cues of other users' activities and areas of focus in shared workspaces.

Live cursors are utilized in applications such as online whiteboards to enable users to see where others are pointing or what they are drawing. They are also used in slideshow applications to enable users to see which slide each member is interacting with.
```

File: examples/spaces-live-cursors/page.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+Live cursors can be implemented using [Ably Spaces](https://ably.com/docs/products/spaces). The Spaces SDK contains a set of purpose-built APIs that abstract away the complexities involved in architecting real-time presence features. It is built on top of Ably's core platform, and so it provides the same performance and scaling guarantees.
```
Comment (m-hulbert) [inline examples/spaces-live-cursors/page.md:9]:
```suggestion
Live cursors are implemented using [Ably Spaces](https://ably.com/docs/products/spaces). The Spaces SDK contains a set of purpose-built APIs that abstract away the complexities involved in managing the state of participants in a collaborative application. It is built on top of Ably's core platform, and so it provides the same performance and scaling guarantees.
```

File: examples/spaces-live-cursors/page.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+// React brief (Only visible if viewing the React example)
+
+Use the following components to add live cursors into an application:
+
+* [SpacesProvider](https://ably.com/docs/spaces/react#spaces-provider): initializes and manages a shared space client instance, passing it down through React context to enable real-time spaces functionality across the application.
+* [SpaceProvider](https://ably.com/docs/spaces/react#spaces-provider): manages the state and functionality of a specific space, providing access to real-time interactions within that space via React context.
+* [useSpace()](https://ably.com/docs/spaces/react#useSpace) hook: a hook to subscribe to the current Space, receive Space state events, and get the current Space instance.
+* [useMembers()](https://ably.com/docs/spaces/react#useMembers) hook: a hook to build live cursors. It retrieves members of the space, including members that have left the space, but have not yet been removed.
+* [useCursors()](https://ably.com/docs/spaces/react#useCursors) hook: a hook to track a member's cursor position and provide a view of all members' cursors within a space.
```
Comment (m-hulbert) [inline examples/spaces-live-cursors/page.md:21]: Code styled please.

**Review by m-hulbert — CHANGES_REQUESTED** [review]
The JavaScript example is only displaying cursors when you hover over the text.

**Comment by GregHolmes** [comment]
> The JavaScript example is only displaying cursors when you hover over the text.

Thanks! I've found the issue and resolved it.

File: examples/spaces-live-cursors/page.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+Live cursors are implemented using [Ably Spaces](https://ably.com/docs/products/spaces). The Spaces SDK contains a set of purpose-built APIs that abstract away the complexities involved in managing the state of participants in a collaborative application. It is built on top of Ably's core platform, and so it provides the same performance and scaling guarantees.
+
+## Resources
+
+// React brief (Only visible if viewing the React example)
+
+Use the following components to add live cursors into an application:
+
+* [`SpacesProvider`](https://ably.com/docs/spaces/react#spaces-provider): initializes and manages a shared space client instance, passing it down through React context to enable real-time spaces functionality across the application.
+* [`SpaceProvider`](https://ably.com/docs/spaces/react#spaces-provider): manages the state and functionality of a specific space, providing access to real-time interactions within that space via React context.
```
Comment (m-hulbert) [inline examples/spaces-live-cursors/page.md:18]: Noticed this after the others: real-time --> realtime

**Review by m-hulbert — APPROVED** [review]
That fixed the JS issue 🚀 

Noticed the same minor real-time typo but LGTM other than that

**Pull request merged**